python/ray/tests/test_network_failure_e2e.py

```diff
@@ -124,6 +125,77 @@ def check_task_pending(n=0):
     wait_for_condition(lambda: check_task_pending(2))
 
 
+head2 = gen_head_node(
```

Comment:
this test seems very useful. wonder if we should extend it to more e2e tests in some stress tests?

Reply:
Yea, I'll use the same pattern for other tests. I'm also exploring chaos test tools for randomly injecting network errors.